src/main/java/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -178,6 +178,18 @@ public TreeReference getOriginalContext() {
         }
     }
 
+    public void enableExpressionCaching() {
+        this.expressionCacher = new InFormExpressionCacher();
+    }
+
+    public boolean expressionCachingEnabled() {
+        return expressionCacher != null;
+    }
+
+    public InFormExpressionCacher expressionCacher() {
```

Comment:
Super nitpicky, but "InForm" feels like an awkward thing for this to be based on v. maybe "MainInstance" (IE: Anything rooted at '/'), just to keep the vocabulary within what is generally referred to in the EvaluationContext

Reply:
Are you referring just to what `InFormExpressionCacher` is called?

Reply:
Yeah. It's just a bit of an odd vocabulary coupling since the EC isn't inherently part of the form.